frmts/gtiff/cogdriver.cpp

```diff
@@ -1369,6 +1369,8 @@ void GDALRegister_COG()
     poDriver->SetMetadataItem( GDAL_DCAP_RASTER, "YES" );
     poDriver->SetMetadataItem( GDAL_DMD_LONGNAME, "Cloud optimized GeoTIFF generator" );
     poDriver->SetMetadataItem( GDAL_DMD_HELPTOPIC, "drivers/raster/cog.html" );
+    poDriver->SetMetadataItem( GDAL_DMD_EXTENSION, "tif" );
```

Comment:
GDAL_DMD_EXTENSION is legacy in that situation and can be removed when there's GDAL_DMD_EXTENSIONS

```suggestion
```